docs/decisions/0002-initial-api-specification.rst

```diff
@@ -0,0 +1,436 @@
+- ``access_policy_uuid`` (POST data, required): The uuid of the policy that allowed the ledger transaction to be created.
+
+Outputs
+-------
+Returns data about the transaction.
+
+::
+
+   {
+       'uuid': 'the-transaction-uuid',
+       'status': 'completed',
+       'idempotency_key': 'the-idempotency-key',
+       'user_id': 54321,
+       'content_key': 'demox_1234+2T2023',
+       'quantity': 19900,
+       'unit': 'USD_CENTS',
+       'reference_id': 1234,
+       'reference_table': 'enrollments',
+       'subsidy_access_policy_uuid': 'a-policy-uuid',
+       'metadata': {...},
+       'created': 'created-datetime',
+       'modified': 'modified-datetime',
+       'reversal': null
+   }
+
+Permissions
+-----------
+
+enterprise_learner
+  NOT ALLOWED
+
+enterprise_admin
+  NOT ALLOWED
+
+openedx_operator
+  If the requesting user is implicitly or explicitly granted this role, they have permissions to create
+  transactions for any active ledgered-subsidy.
+
+POST enterprise subsidy transaction reversal
+============================================
+**/api/v1/subsidies/[subsidy-uuid]/transactions/[transaction-uuid]/reverse**
+
+Reverse a subsidy transaction in the subsidy's ledger.
+Only service users (those with ``openedx_operator`` role assignment) can do this.
+A possible side-effect of a successful POST request here is the downgrading of any associated ``verified``
+enrollment records to the ``audit`` mode.
+The subsidy-service should determine the price of the content before processing the reversal.
+
+Inputs
+------
+
+- ``subsidy-uuid`` (URL path, required): The uuid (primary key) of the subsidy for which a transaction should be reversed.
+- ``transaction-uuid`` (URL path, required): The uuid (primary key) of the transaction to reverse.
+
+Outputs
+-------
+Returns data about the transaction and its reversal.
+
+::
+
+   {
+       'uuid': 'the-transaction-uuid',
+       'status': 'completed',
+       'idempotency_key': 'the-idempotency-key',
+       'user_id': 54321,
+       'content_key': 'demox_1234+2T2023',
+       'quantity': 19900,
+       'unit': 'USD_CENTS',
+       'reference_id': 1234,
+       'reference_table': 'enrollments',
+       'subsidy_access_policy_uuid': 'a-policy-uuid',
+       'metadata': {...},
+       'created': 'created-datetime',
+       'modified': 'modified-datetime',
+       'reversal': {
+           'idempotency_key': 'the-reversal-idempotency-key',
+           'created': 'created-datetime',
+           'modified': 'modified-datetime',
+           'quantity': -19900,
+           'metadata': null,
+       }
+   }
+
+Permissions
+-----------
+
+enterprise_learner
+  NOT ALLOWED
+
+enterprise_admin
+  NOT ALLOWED
+
+openedx_operator
+  If the requesting user is implicitly or explicitly granted this role, they have permissions to reverse
+  transactions for any active ledgered-subsidy.
+
+POST can redeem in subsidy
+==========================
+**/api/v1/subsidies/[subsidy-uuid]/can_redeem/**
```

Comment:
does this POST endpoint have a side-effect?

Reply:
yeah... Should this be a GET?